File: app/src/main/java/eu/kanade/mangafeed/data/database/models/Category.java
package eu.kanade.mangafeed.data.database.models;

import com.pushtorefresh.storio.sqlite.annotations.StorIOSQLiteColumn;
import com.pushtorefresh.storio.sqlite.annotations.StorIOSQLiteType;

import java.io.Serializable;

import eu.kanade.mangafeed.data.database.tables.CategoryTable;

@StorIOSQLiteType(table = CategoryTable.TABLE)
public class Category implements Serializable {

    @StorIOSQLiteColumn(name = CategoryTable.COLUMN_ID, key = true)
    public Integer id;

    @StorIOSQLiteColumn(name = CategoryTable.COLUMN_NAME)
    public String name;

    @StorIOSQLiteColumn(name = CategoryTable.COLUMN_ORDER)
    public int order;

    @StorIOSQLiteColumn(name = CategoryTable.COLUMN_FLAGS)
    public int flags;

    public Category() {}

    public static Category create(String name) {
        Category c = new Category();
        c.name = name;
        return c;
    }

    public static Category createDefault() {
        Category c = create("Default");
        c.id = 0;
        return c;
    }
}

File: app/src/main/java/eu/kanade/domain/category/model/Category.kt
package eu.kanade.domain.category.model

import java.io.Serializable

data class Category(
    val id: Long,
    val name: String,
    val order: Long,
    val flags: Long,
) : Serializable {

    val isSystemCategory: Boolean = id == UNCATEGORIZED_ID

    companion object {

        const val UNCATEGORIZED_ID = 0L
    }
}

internal fun List<Category>.anyWithName(name: String): Boolean {
    return any { name == it.name }
}
